server/jetty/src/main/java/io/deephaven/server/jetty/CacheFilter.java

```diff
@@ -0,0 +1,36 @@
+package io.deephaven.server.jetty;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.Date;
+
+public class CacheFilter implements Filter {
+    public static final int YEAR_IN_SECONDS = 365 * 24 * 60 * 60;
+
+    @Override
+    public void doFilter(final ServletRequest request,
+            final ServletResponse response,
+            final FilterChain filterChain)
+            throws IOException, ServletException {
+        final HttpServletResponse httpResponse = (HttpServletResponse) response;
+
+        // set expiry to one year in the future
+        final Calendar calendar = Calendar.getInstance();
+        calendar.setTime(new Date());
+        calendar.add(Calendar.YEAR, 1);
+        httpResponse.setDateHeader("Expires", calendar.getTime().getTime());
+        // Note: immutable tells firefox to never revalidate as data will never change
+        httpResponse.setHeader("Cache-control", "max-age=" + YEAR_IN_SECONDS + ", public, immutable");
+        httpResponse.setHeader("Pragma", "");
+
+        filterChain.doFilter(request, response);
```

Comment:
Take a look at where we plug in the CacheFilter over in JettyBackedGrpcServer - those files in the static dir are all generated with their hash as part of their name, so that without a name change, the contents are (in theory) unchanged.

These filters and their configuration are the same as in DHE - to a fault, I didn't correct the path to be /ide/...